server/src/main/java/org/opensearch/indices/replication/OngoingSegmentReplications.java

```diff
@@ -269,6 +269,9 @@ private void cancelHandlers(Predicate<? super SegmentReplicationSourceHandler> p
             .filter(predicate)
             .map(SegmentReplicationSourceHandler::getAllocationId)
             .collect(Collectors.toList());
+        if (allocationIds.size() == 0) {
```

Comment:
Nit. Why we need this check here? if list is empty we will anyhow return right?

Reply:
This is to prevent execution of follow up log statement